Include/internal/pycore_object.h

```diff
@@ -220,7 +224,46 @@ _Py_DECREF_SPECIALIZED(PyObject *op, const destructor destruct)
 static inline void
 _Py_DECREF_NO_DEALLOC(PyObject *op)
 {
-    Py_DECREF(op);
+    if (_Py_IsImmortal(op)) {
+        return;
+    }
+
+    if (_Py_IsOwnedByCurrentThread(op)) {
+        uint32_t refcount = _Py_atomic_load_uint32_relaxed(&op->ob_ref_local);
+        assert(refcount != 0);
+        refcount--;
+        _Py_atomic_store_uint32_relaxed(&op->ob_ref_local, refcount);
+        if (refcount == 0) {
```

Comment:
Similar question, do we have to handle local refcounting zero case from _Py_DECREF_NO_DEALLOC or it can be handled as deferred from somewhere?

Reply:
Yes, you need `_Py_MergeZeroLocalRefcount`. You can't defer it -- that would break a bunch of invariants. For example, the same thread may try calling `Py_DECREF` again leading to a negative local refcount.